### PR TITLE
support binary static content in JTE

### PR DESCRIPTION
**Pull request by edward3h**
Update JTE to latest version.

Support for 'binary static content' in JTE. This pre-converts Strings to UTF-8 bytes to reduce work during rendering.

### CODE REVIEW

**Review by sdelamo — COMMENTED** [review]
@edward3h Thanks for the contribution. 

could you  add more tests? Could you tests it works both for `binaryStaticContent` true and false. 

not important but we prefer not to use wildcard imports if possible. 

Not sure if it make sense for JTE but since `JteWritable` implements `Writable` it could be possible to just use it as a controller method return. Maybe you could document that. See `RockerWritable`.

**Comment by edward3h** [comment]
I've made the tests run both ways with the binary output turned on and off. Also changed star imports back to single classes.

I'd like to do something similar to the `RockerWritable` example, but I'm not satisfied with the JTE API for it at the moment. I'll come back to that some time.

**Review by graemerocher — APPROVED** [review]

File: views-jte/src/main/java/io/micronaut/views/jte/JteViewsRenderer.java [hunk 1]
```diff
@@ -58,27 +57,36 @@ protected JteViewsRenderer(
...
+        return output;
+    }
+
+    /**
+     * No longer used. Retained for binary compatibility.
+     *
+     * @param out output writer
+     * @return JTE output
+     */
+    @NonNull
```
Comment (sdelamo) [inline views-jte/src/main/java/io/micronaut/views/jte/JteViewsRenderer.java:89]: annotate with `@Deprecated`

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: views-jte/src/main/java/io/micronaut/views/jte/JteViewsRenderer.java (hunk 1 above)
Comment (sdelamo) [reply views-jte/src/main/java/io/micronaut/views/jte/JteViewsRenderer.java:89]: resolved via https://github.com/micronaut-projects/micronaut-views/pull/428

**Pull request merged**